### PR TITLE
Using individual lodash modules

### PR DESCRIPTION
**Pull request by amolde**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
-var _       = require("lodash"),
+const _assign = require("lodash.assign"),
```
Comment (afrasso) [inline src/index.js]: I'd rather not use es6 here to remain as compatible as possible with other frameworks. Also, I'd prefer the ordering of the requires to be alphabetical.

File: src/index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+    _toArray = require("lodash.toarray"),
+    _forEach = require("lodash.foreach"),
+    _concat = require("lodash.concat"),
+    _get = require("lodash.get"),
```
Comment (afrasso) [inline src/index.js]: I also don't love the underscores everywhere. The lodash docs seem to just recommend just calling it "get": https://www.npmjs.com/package/lodash.get.

File: src/index.js [hunk 3]
```diff
@@ -11,32 +15,32 @@ var RequestPromiseMiddlewareFramework = function(rp) {
...
 
 RequestPromiseMiddlewareFramework.prototype.getMiddlewareEnabledRequestPromise = function() {
   var me = this;
   var intercept = function(options) {
-    var resolveWithFullResponse = _.get(options, "resolveWithFullResponse");
-    var middleware = _.concat(me.middleware, me.initialMiddleware);
-    var next = function(_options, _callback) {
+    var resolveWithFullResponse = _get(options, "resolveWithFullResponse");
+    var middleware = _concat(me.middleware, me.initialMiddleware);
+    var next = function(__options, __callback) {
```
Comment (afrasso) [inline src/index.js]: Revert this variables back to single underscore.

File: src/index.js [hunk 4]
```diff
@@ -1,6 +1,10 @@
-var _       = require("lodash"),
-    assert  = require("assert-plus"),
-    Promise = require("bluebird");
+var assign   = require("lodash.assign"),
+    concat   = require("lodash.concat"),
+    forEach  = require("lodash.foreach"),
+    get      = require("lodash.get"),
+    toArray  = require("lodash.toarray"),
+    assert   = require("assert-plus"),
+    Promise  = require("bluebird");
```
Comment (afrasso) [inline src/index.js]: Sorry -- I meant all of these should be alphabetized, not just within the lodash modules. So it should be:

```
var assert  = require("assert-plus"),
    assign  = require("lodash.assign"),
    concat  = require("lodash.concat"),
    forEach = require("lodash.foreach"),
    get     = require("lodash.get"),
    Promise = require("bluebird"),
    toArray = require("lodash.toarray");
```

Comment (amolde) [reply src/index.js]: Oops! I sorted them before removing the lodash (__) prefix and hence the weird sort order.

**Pull request merged**